http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -252,19 +256,17 @@ private void cleanupRequest(ChannelHandlerContext ctx, NettyHttpRequest request)
         try {
             request.release();
         } finally {
-            ctx.executor().execute(() -> {
-                try {
-                    beanContext.publishEvent(
-                            new HttpRequestTerminatedEvent(
-                                    request
-                            )
-                    );
-                } catch (Exception e) {
-                    if (LOG.isErrorEnabled()) {
-                        LOG.error("Error publishing request terminated event: " + e.getMessage(), e);
+            if (terminateEventPublisher != ApplicationEventPublisher.NO_OP) {
```

Comment:
Perhaps add a method to event publisher to determine if there are listeners instead. It would be pretty difficult for users to find out this solution

Reply:
This is a performance hack but I guess it can be more official

Reply:
Added `ApplicationEventPublisher#noOp()`